3party/protobuf/CMakeLists.txt

```diff
@@ -1,43 +1,38 @@
 project(protobuf)
 
-set(SRC
-  config.h
-  protobuf/src/google/protobuf/arena.cc
-  protobuf/src/google/protobuf/extension_set.cc
-  protobuf/src/google/protobuf/generated_message_util.cc
-  protobuf/src/google/protobuf/io/coded_stream.cc
-  protobuf/src/google/protobuf/io/zero_copy_stream.cc
-  protobuf/src/google/protobuf/io/zero_copy_stream_impl_lite.cc
-  protobuf/src/google/protobuf/message_lite.cc
-  protobuf/src/google/protobuf/repeated_field.cc
-  protobuf/src/google/protobuf/stubs/atomicops_internals_x86_gcc.cc
-  protobuf/src/google/protobuf/stubs/bytestream.cc
-  protobuf/src/google/protobuf/stubs/common.cc
-  protobuf/src/google/protobuf/stubs/int128.cc
-  protobuf/src/google/protobuf/stubs/once.cc
-  protobuf/src/google/protobuf/stubs/status.cc
-  protobuf/src/google/protobuf/stubs/statusor.cc
-  protobuf/src/google/protobuf/stubs/stringpiece.cc
-  protobuf/src/google/protobuf/stubs/stringprintf.cc
-  protobuf/src/google/protobuf/stubs/structurally_valid.cc
-  protobuf/src/google/protobuf/stubs/strutil.cc
-  protobuf/src/google/protobuf/stubs/time.cc
-  protobuf/src/google/protobuf/wire_format_lite.cc
-)
+# Disable all compile-time warnings for protobuf
+add_compile_options(-w)
 
-add_library(${PROJECT_NAME} ${SRC})
+# protobuf config
+set(protobuf_LIB_NAME libprotobuf-lite)
+set(protobuf_VERSION "23.4")
```

Comment:
```suggestion
set(protobuf_VERSION "24.4")
```

Reply:
@AndrewShkrob I don't like updating it manually every time. Can it be automated? Is it needed at all?

Reply:
This line is required:
```cmake
set_target_properties(libprotobuf-lite PROPERTIES
    VERSION ${protobuf_VERSION}
    OUTPUT_NAME ${LIB_PREFIX}protobuf-lite
    DEBUG_POSTFIX "${protobuf_DEBUG_POSTFIX}"
    # For -fvisibility=hidden and -fvisibility-inlines-hidden
    C_VISIBILITY_PRESET hidden
    CXX_VISIBILITY_PRESET hidden
    VISIBILITY_INLINES_HIDDEN ON
)
```

I think it's not a big deal to update the version each time you update the version... 
But if you don't like it, you can set any dummy value here, for example, `"OrganicMaps"`, or the commit hash of the protobuf repo.

There is also another way. The version of protobuf is stored in `3party/protobuf/protobuf/version.json`:
```json
{
    "24.x": {
        "protoc_version": "24.4",
        "lts": false,
        "date": "2023-10-03",
        "languages": {
            "cpp": "4.24.4",
            "csharp": "3.24.4",
            "java": "3.24.4",
            "javascript": "3.24.4",
            "objectivec": "3.24.4",
            "php": "3.24.4",
            "python": "4.24.4",
            "ruby": "3.24.4"
        }
    }
}
```

We can write a script to fetch `"protoc_version"`, but I think it's an overcomplication.